### PR TITLE
feat: add image converter

### PR DESCRIPTION
**Pull request by gaston-flores**
## Type of Pull Request

- [ ] Bug fix
- [x] Enhancement

Related Issue #s or links (if any): 

#28 

## Description of Changes

This PR adds an image converter tool:

image

https://user-images.githubusercontent.com/18354222/194715964-727e53a3-68e2-46c4-8a03-c9cb90285afb.mp4

Some notes:

- The current version doesn't show a live preview of your changes, only the original image. A live preview is possible, but the image generation is not very performant. My machine struggled to keep up as I made changes. 
- There's only error handling for cases where the user doesn't upload an image, but I don't validate if the upload is a valid image.
- I'm waiting for feedback on the changes before adding tests.

### CODE REVIEW

File: pages/image-converter.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  );
+}
+
+function formatSliderLabel(value: number) {
+  return `${Math.round(value * 100)}%`;
+}
+
+const PreviewImage = styled('img')({});
+
+export default function JsonPage() {
```
Comment (dlford) [inline pages/image-converter.tsx:49]: Minor, but we should rename this function

File: pages/image-converter.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  const [file, setFile] = useState<{ raw: File; imageSrc: string }>();
+  const [fileType, setFileType] = useState<FileType>(FileType.WEBP);
+  const [quality, setQuality] = useState(0.7);
```
Comment (dlford) [inline pages/image-converter.tsx:52]: I think we could use `useLocalState` for `fileType` and `quality` to store the last used settings. Feel free to skip if you don't think it's appropriate.

File: pages/image-converter.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  WEBP = 'image/webp',
+}
+
+const FILE_TYPE_EXTENSIONS = {
+  [FileType.PNG]: '.png',
+  [FileType.JPG]: '.jpg',
+  [FileType.WEBP]: '.webp',
+};
+
+const FILE_TYPE_OPTIONS = [
```
Comment (dlford) [inline pages/image-converter.tsx:30]: No avif option, maybe browser support isn't quite there yet, if that's the case there's nothing we can do right now.

File: pages/image-converter.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    // Bail immediately if the user canceled
+    if (
+      !event.currentTarget.files ||
+      event.currentTarget.files.length === 0
+    ) {
+      return;
+    }
+    const eventFile = event.currentTarget.files[0];
+    const reader = new FileReader();
+    console.log(eventFile);
```
Comment (dlford) [inline pages/image-converter.tsx:65]: Please remove the console log here

File: pages/image-converter.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+              </RadioGroup>
+            </FormControl>
+            <Box sx={{ my: 3 }}>
+              <Typography
+                id='quality-slider'
+                gutterBottom
+              >
+                Quality
+              </Typography>
+              <Slider
```
Comment (dlford) [inline pages/image-converter.tsx:172]: The quality slider doesn't seem to do anything for PNG images, can we disable that input when PNG is selected?

**Review by dlford — CHANGES_REQUESTED** [review]
@gaston-flores great job on this! Just a few minor tweaks if it's not too much trouble.

File: pages/image-converter.tsx (hunk 3 above)
Comment (gaston-flores) [reply pages/image-converter.tsx:30]: Yes, unfortunately Chrome/Firefox still don't support AVIF in `canvas` 😢

File: pages/image-converter.tsx (hunk 4 above)
Comment (gaston-flores) [reply pages/image-converter.tsx:65]: My bad! Do you think it'd be helpful to enable [`no-console`](https://eslint.org/docs/latest/rules/no-console)?

Comment (dlford) [reply pages/image-converter.tsx:65]: Yes, it was enabled before but I made some changes and must've clobbered that lol

File: pages/image-converter.tsx (hunk 3 above)
Comment (dlford) [reply pages/image-converter.tsx:30]: No worries, I'll make a new issue to solve that when it is supported

**Review by dlford — CHANGES_REQUESTED** [review]
@gaston-flores looks great! Just need some tests and we are good to merge.

**Pull request merged**